internal/test/wmcb_test.go

```diff
@@ -0,0 +1,234 @@
+var framework = &testFramework{remoteDir: "C:\\Temp"}
+
+// binaryToBeTransferred holds the binary that needs to be transferred to the Windows VM
+// TODO: Make this an array later with a comma separated values for more binaries to be transferred
+var binaryToBeTransferred = flag.String("binaryToBeTransferred", "",
+	"Absolute path of the binary to be transferred")
+
+// setup sets up the initial test cluster for the Windows
+// TODO: move this to return error and do assertions there
+func (f *testFramework) setup() {
+	if err := framework.createWindowsVM(); err != nil {
+		log.Fatalf("failed to create Windows VM: %v", err)
+	}
+	// TODO: Add some options to skip certain parts of the test
+	if err := framework.setupWinRMClient(); err != nil {
+		log.Fatalf("failed to setup winRM client for the Windows VM: %v", err)
+	}
+	// Wait for some time before starting configuring of ssh server. This is to let sshd service be available
+	// in the list of services
+	// TODO: Parse the output of the `Get-Service sshd, ssh-agent` on the Windows node to check if the windows nodes
+	// has those services present
+	time.Sleep(time.Minute)
+	if err := framework.configureOpenSSHServer(); err != nil {
+		log.Fatalf("failed to configure OpenSSHServer on the Windows VM: %v", err)
+	}
+	if err := framework.createRemoteDir(); err != nil {
+		log.Fatalf("failed to create remote dir with error: %v", err)
+	}
+	if err := framework.getSSHClient(); err != nil {
+		log.Fatalf("failed to get ssh client for the Windows VM created: %v", err)
+	}
+}
+
+// createWindowsVM spins up the Windows VM in the given cloud provider and gives us the credentials to access the
+// windows VM created
+func (f *testFramework) createWindowsVM() error {
+	kubeconfig := os.Getenv("KUBECONFIG")
+	awsCredentials := os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
+	artifactDir := os.Getenv("ARTIFACT_DIR")
+	privateKeyPath := os.Getenv("KUBE_SSH_KEY_PATH")
+	// Use Windows 2019 server image with containers in us-east1 zone for CI testing.
+	// TODO: Move to environment variable that can be fetched from the cloud provider
+	imageID := "ami-0b8d82dea356226d3"
+	instanceType := "m4.large"
+	sshKey := "libra"
+	cloud, err := cloudprovider.CloudProviderFactory(kubeconfig, awsCredentials, "default", artifactDir,
+		imageID, instanceType, sshKey, privateKeyPath)
+	if err != nil {
+		return fmt.Errorf("error instantiating cloud provider %v", err)
+	}
+	f.cloudProvider = cloud
+	credentials, err := cloud.CreateWindowsVM()
+	if err != nil {
+		return fmt.Errorf("error creating Windows VM: %v", err)
+	}
+	f.credentials = credentials
+	return nil
+}
+
+// setupWinRMClient sets up the winrm client to be used while accessing Windows node
+func (f *testFramework) setupWinRMClient() error {
+	host := f.credentials.GetIPAddress()
+	password := f.credentials.GetPassword()
+
+	endpoint := winrm.NewEndpoint(host, winRMPort, true, true, nil, nil, nil, 0)
+	winrmClient, err := winrm.NewClient(endpoint, user, password)
```

Comment:
Would it make sense to get rid of `winrmClient`, and instead set it to `f.winrmClient`?